x/ecocredit/simulation/operations.go

```diff
@@ -0,0 +1,595 @@
+		for _, class := range classes {
+			if contains(class.Issuers, issuer.Address.String()) {
+				classID = class.ClassId
+				break
+			}
+		}
+
+		if classID == "" {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgCreateBatch, "don't have permission to create credit batch"), nil, nil
+		}
+
+		issuerAcc := ak.GetAccount(sdkCtx, issuer.Address)
+		spendable := bk.SpendableCoins(sdkCtx, issuerAcc.GetAddress())
+
+		now := ctx.BlockTime()
+		tenHours := now.Add(10 * time.Hour)
+		msg := &ecocredit.MsgCreateBatch{
+			Issuer:          issuer.Address.String(),
+			ClassId:         classID,
+			Issuance:        generateBatchIssuence(r, accs),
+			StartDate:       &now,
+			EndDate:         &tenHours,
+			Metadata:        []byte(simtypes.RandStringOfLength(r, 10)),
+			ProjectLocation: "AB-CDE FG1 345",
+		}
+
+		txCtx := simulation.OperationInput{
+			R:               r,
+			App:             app,
+			TxGen:           simappparams.MakeTestEncodingConfig().TxConfig,
+			Cdc:             nil,
+			Msg:             msg,
+			MsgType:         msg.Type(),
+			Context:         sdkCtx,
+			SimAccount:      issuer,
+			AccountKeeper:   ak,
+			Bankkeeper:      bk,
+			ModuleName:      ecocredit.ModuleName,
+			CoinsSpentInMsg: spendable,
+		}
+
+		return GenAndDeliverTxWithRandFees(txCtx)
+	}
+}
+
+// SimulateMsgSend generates a MsgSend with random values.
+func SimulateMsgSend(ak exported.AccountKeeper, bk exported.BankKeeper,
+	qryClient ecocredit.QueryClient) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, sdkCtx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+
+		ctx := regentypes.Context{Context: sdkCtx}
+		class, op, err := getRandomClass(ctx, r, qryClient, TypeMsgSend)
+		if class == nil {
+			return op, nil, err
+		}
+
+		batch, op, err := getRandomBatchFromClass(ctx, r, qryClient, TypeMsgSend, class.ClassId)
+		if batch == nil {
+			return op, nil, err
+		}
+
+		balres, err := qryClient.Balance(ctx, &ecocredit.QueryBalanceRequest{
+			Account:    batch.Issuer,
+			BatchDenom: batch.BatchDenom,
+		})
+		if err != nil {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSend, err.Error()), nil, err
+		}
+
+		tradableBalance, err := math.NewNonNegativeDecFromString(balres.TradableAmount)
+		if err != nil {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSend, err.Error()), nil, err
+		}
+
+		retiredBalance, err := math.NewNonNegativeDecFromString(balres.RetiredAmount)
+		if err != nil {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSend, err.Error()), nil, err
+		}
+
+		if tradableBalance.IsZero() || retiredBalance.IsZero() {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSend, "balance is zero"), nil, nil
+		}
+
+		recipient, _ := simtypes.RandomAcc(r, accs)
+		if batch.Issuer == recipient.Address.String() {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSend, "sender & reciever are same"), nil, nil
```

Comment:
let's use consistent terminology
```suggestion
			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgSend, "sender & recipient are same"), nil, nil
```